docs/specs/2fa-grace-period.md

```diff
@@ -0,0 +1,77 @@
+# 2FA Grace Period
+
+- [Abstract](#abstract)
+- [Configuration](#configuration)
+  - [Global Grace Period](#global-grace-period)
+  - [Per-user Grace Period](#per-user-grace-period)
+  - [Customized Authentication Flow](#customized-authentication-flow)
+
+## Abstract
+
+In order to enforce 2FA, we will provide a grace period for users to enroll in 2FA after they have signed up.
+
+After the grace period, the user will be required to enroll in 2FA before they can sign in.
+
+## Configuration
+
+By default, no grace period is provided. User without 2FA must contact admin to login after [2FA mode](./user-model.md#secondary-authenticator) is set to `required`.
+
+For new users, the grace period starts from the time the user is created, whereas for existing users, the grace period starts from when the grace period is enabled.
+
+### Global Grace Period
+
+Global grace period can be enabled for forcing users to enroll in 2FA upon login.
+
+```yaml
+authentication:
+  secondary_authentication_mode: "required"
+  secondary_authentication_rollout:
+    global_grace_period_enabled_since: "2021-01-01T00:00:00Z"
+    global_grace_period_days: 30
```

Comment:
```
secondary_authentication_grace_period:
  enabled: true
```

This means indefinite grace period.

```
secondary_authentication_grace_period:
  enabled: true
  end_at: "2021-01-01T00:00:00Z"
```

This means grace period is enabled, and it ends at 2021-01-01T00:00:00Z

---

Remove `global_grace_period_days`

Reply:
Updated.